libs/blocks/form/form.js

```diff
@@ -0,0 +1,378 @@
+    if (fd.Default === text) select.value = text;
+  });
+  if (fd.Required === 'x') {
+    select.setAttribute('required', 'required');
+  }
+  return select;
+}
+
+function constructPayload(form) {
+  const payload = {};
+  [...form.elements].forEach((fe) => {
+    if (fe.type.match(/(?:checkbox|radio)/)) {
+      if (fe.checked) {
+        payload[fe.name] = payload[fe.name] ? `${fe.value}, ${payload[fe.name]}` : fe.value;
+      } else if (fe.closest('.group-container').classList.contains('required')) {
+        payload[fe.name] = 'required-not-checked';
+      }
+    } else if (fe.id) {
+      payload[fe.id] = fe.value;
+    }
+  });
+  return payload;
+}
+
+async function submitForm(form) {
+  const payload = constructPayload(form);
+  const keys = Object.keys(payload);
+  payload.timestamp = new Date().toJSON();
+  for (const key of keys) {
+    if (payload[key] === 'required-not-checked') {
+      const el = form.querySelector(`input[name="${key}"]`);
+      el.setCustomValidity('This box must be checked');
+      el.reportValidity();
+      const cb = () => {
+        el.setCustomValidity('');
+        el.reportValidity();
+        el.removeEventListener('input', cb);
+      };
+      el.addEventListener('input', cb);
+      return false;
+    }
+    payload[key] = sanitizeComment(payload[key]);
+  }
+  /* c8 ignore next 7 */
+  const resp = await fetch(form.dataset.action, {
+    method: 'POST',
+    cache: 'no-cache',
+    headers: { 'Content-Type': 'application/json' },
+    body: JSON.stringify({ data: payload }),
+  });
+  await resp.text();
+  return payload;
+}
+
+function clearForm(form) {
+  [...form.elements].forEach((fe) => {
+    if (fe.type.match(/(?:checkbox|radio)/)) {
+      fe.checked = false;
+    } else {
+      fe.value = '';
+    }
+  });
+}
+
+function createButton(fd, thankYou) {
+  const button = document.createElement('button');
+  button.textContent = fd.Label;
+  button.classList.add('button');
+  if (fd.Type === 'submit') {
+    button.addEventListener('click', async (event) => {
+      const form = button.closest('form');
+      if (form.checkValidity()) {
+        event.preventDefault();
+        button.setAttribute('disabled', '');
+        const submission = await submitForm(form);
+        button.removeAttribute('disabled');
+        if (!submission) return;
+        clearForm(form);
+        const handleThankYou = thankYou.querySelector('a') ? thankYou.querySelector('a').href : thankYou.innerHTML;
+        if (!thankYou.innerHTML.includes('href')) {
+          const thanksText = document.createElement('h4');
+          thanksText.className = 'thank-you';
+          thanksText.textContent = handleThankYou;
+          form.append(thanksText);
+          setTimeout(() => thanksText.remove(), 2000);
+          /* c8 ignore next 3 */
+        } else {
+          window.location.href = handleThankYou;
+        }
+      }
+    });
+  } else if (fd.Type === 'clear') {
+    button.classList.add('outline');
```

Comment:
Nit. Does this need to be an else if? 
If submit do submit stuff.
If clear do clear stuff.